crates/core/src/factories/stub_transport.rs

```diff
@@ -0,0 +1,341 @@
+            task_list.clone(),
+            handler,
+            this_url,
+            cmd_send.clone(),
+            cmd_recv,
+        ));
+
+        let out: DynTxImp = Arc::new(Self {
+            task_list,
+            cmd_send,
+        });
+
+        out
+    }
+}
+
+impl TxImp for StubTransport {
+    fn disconnect(
+        &self,
+        peer: Url,
+        payload: Option<(String, bytes::Bytes)>,
+    ) -> BoxFut<'_, ()> {
+        Box::pin(async move {
+            let (s, r) = tokio::sync::oneshot::channel();
+            if self
+                .cmd_send
+                .send(Cmd::Disconnect(peer, payload, s))
+                .is_ok()
+            {
+                let _ = r.await;
+            }
+        })
+    }
+
+    fn send(&self, peer: Url, data: bytes::Bytes) -> BoxFut<'_, K2Result<()>> {
+        Box::pin(async move {
+            let (s, r) = tokio::sync::oneshot::channel();
+            match self.cmd_send.send(Cmd::Send(peer, data, s)) {
+                Err(_) => Err(K2Error::other("Connection Closed")),
+                Ok(_) => match r.await {
+                    Ok(r) => r,
+                    Err(_) => Err(K2Error::other("Connection Closed")),
+                },
+            }
+        })
+    }
+}
+
+type Res = tokio::sync::oneshot::Sender<K2Result<()>>;
+type CmdSend = tokio::sync::mpsc::UnboundedSender<Cmd>;
+type CmdRecv = tokio::sync::mpsc::UnboundedReceiver<Cmd>;
+type DataSend = tokio::sync::mpsc::UnboundedSender<(bytes::Bytes, Res)>;
+type DataRecv = tokio::sync::mpsc::UnboundedReceiver<(bytes::Bytes, Res)>;
+type ConSend = tokio::sync::mpsc::UnboundedSender<(Url, DataSend, DataRecv)>;
+type ConRecv = tokio::sync::mpsc::UnboundedReceiver<(Url, DataSend, DataRecv)>;
+
+struct DropSend {
+    send: DataSend,
+    handler: Arc<TxImpHnd>,
+    peer: Url,
+    reason: Option<String>,
+}
+
+impl Drop for DropSend {
+    fn drop(&mut self) {
+        self.handler
+            .peer_disconnect(self.peer.clone(), self.reason.take());
+    }
+}
+
+impl DropSend {
+    fn new(send: DataSend, handler: Arc<TxImpHnd>, peer: Url) -> Self {
+        Self {
+            send,
+            handler,
+            peer,
+            reason: None,
+        }
+    }
+}
+
+enum Cmd {
+    RegCon(Url, DataSend, DataRecv),
+    InData(Url, bytes::Bytes, Res),
+    Disconnect(Url, Option<(String, bytes::Bytes)>, Res),
+    Send(Url, bytes::Bytes, Res),
+}
+
+async fn cmd_task(
+    task_list: Arc<Mutex<tokio::task::JoinSet<()>>>,
+    handler: Arc<TxImpHnd>,
+    this_url: Url,
+    cmd_send: CmdSend,
+    mut cmd_recv: CmdRecv,
+) {
+    let mut con_pool = HashMap::new();
+
+    while let Some(cmd) = cmd_recv.recv().await {
+        match cmd {
+            Cmd::RegCon(url, data_send, mut data_recv) => {
+                match handler.peer_connect(url.clone()) {
+                    Err(_) => continue,
+                    Ok(preflight) => {
+                        let (s, _) = tokio::sync::oneshot::channel();
```

Comment:
What's the point of this oneshot channel if there's no receiver?

Reply:
The command variant requires the sender, we just don't care about awaiting the receiver in this case.